pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -95,7 +95,24 @@ func (d *DependenciesDistributor) OnAdd(obj interface{}) {
 }
 
 // OnUpdate handles object update event and push the object to queue.
-func (d *DependenciesDistributor) OnUpdate(_, newObj interface{}) {
+func (d *DependenciesDistributor) OnUpdate(oldObj, newObj interface{}) {
+	unstructuredOldObj, err := helper.ToUnstructured(oldObj)
+	if err != nil {
+		klog.Errorf("Failed to transform oldObj, error: %v", err)
+		return
+	}
+
+	unstructuredNewObj, err := helper.ToUnstructured(newObj)
+	if err != nil {
+		klog.Errorf("Failed to transform newObj, error: %v", err)
+		return
+	}
+
+	if !detector.SpecificationChanged(unstructuredOldObj, unstructuredNewObj) {
+		klog.V(4).Infof("Ignore update event of object (kind=%s, %s/%s) as specification no change", unstructuredOldObj.GetKind(), unstructuredOldObj.GetNamespace(), unstructuredOldObj.GetName())
```

Comment:
how about `klog.V(4).Infof("Ignore update event of object (%s, kind=%s, %s/%s) as specification no change", unstructuredOldObj.APIVersion, unstructuredOldObj.Kind, names.NamespacedKey(unstructuredOldObj.Namespace, unstructuredOldObj.Name))` ?